### PR TITLE
Add post audit withdrawals contract, refactor staking naming

### PR DESCRIPTION
**Pull request by drinkius**

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -11,8 +11,8 @@ contract Staking is InitializableOwnable, IStaking {
 
     struct UserInfo {
         uint amount;
-        uint rewardDebt;
-        uint accumulatedReward;
+        uint rewardAccountedForHarevest;
```
Comment (sapranidi) [inline contracts/Staking.sol]: Minor typo here, the var should be "rewardAccountedForHarvest"

Comment (drinkius) [reply contracts/Staking.sol]: Fixed

File: hardhat.config.ts [hunk 2]
```diff
@@ -25,19 +28,29 @@ module.exports = {
       url: "http://127.0.0.1:8545",
     },
     mainnet: {
-      url: `https://mainnet.infura.io/v3/${INFURA_API_KEY}`,
+      url: "https://rinkeby.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161",
```
Comment (sapranidi) [inline hardhat.config.ts]: @drinkius Is this your Infura api key? Even though it's for Rinkeby, it's better to have this removed

Comment (drinkius) [reply hardhat.config.ts]: Not mine, publicly available in databases online. Added it this way to remove hassle for everyone 

Source:
https://rpc.info/

**Review by tornadofsoulz — APPROVED** [review]

**Pull request merged**